api-service/webapp-demo/src/app/test_question_filter.component.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, Injectable, ViewChild } from '@angular/core';
+import { ComponentFixture, TestBed, async, inject } from '@angular/core/testing';
+import { ActivatedRoute } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { FormsModule, ReactiveFormsModule }   from '@angular/forms';
+import {
+  HttpClientTestingModule,
+  HttpTestingController
+} from '@angular/common/http/testing';
+import { MatButtonModule, MatCheckboxModule } from '@angular/material';
+import { By } from '@angular/platform-browser';
+import { BaseJobComponent } from './base_job.component';
+import { TestQuestionFilterComponent } from './test_question_filter.component';
+import { CrowdSourceApiService } from './crowd_source_api.service';
+import { ActivatedRouteStub, setupQuestionMocks } from './test_util';
+
+@Component({
+  selector: 'base-job-test',
+  template: `<test-question-filter></test-question-filter>`
+})
+class TestQuestionFilterTestComponent {
+  @ViewChild(TestQuestionFilterComponent)
+  testQuestionFilterComponent: TestQuestionFilterComponent;
+}
+
+let activatedRoute: ActivatedRouteStub;
+
+describe('TestQuestionFilterComponent tests', () => {
+  beforeEach(async(() => {
+    activatedRoute = new ActivatedRouteStub();
+    activatedRoute.testParams = {};
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        FormsModule,
+        MatButtonModule,
+        MatCheckboxModule,
+        ReactiveFormsModule,
+        RouterTestingModule.withRoutes(
+          [
+              {
+                path: 'test_question_filter',
+                component: TestQuestionFilterComponent
+              },
+              {
+                path: 'test_question_filter/:customClientJobKey',
+                component: TestQuestionFilterComponent
+              },
+          ]
+        )
+      ],
+      declarations: [
+        TestQuestionFilterTestComponent,
+        TestQuestionFilterComponent,
+      ],
+      providers: [
+        {provide: ActivatedRoute, useValue: activatedRoute},
+        CrowdSourceApiService,
+      ]
+    }).compileComponents();
+  }));
+
+  it('Builds yes answer', async((inject([HttpTestingController], (httpMock: HttpTestingController) => {
+    activatedRoute.testParams = {
```

Comment:
Each of the 3 test cases has a lot of repeated code, it might be good to refactor this into some shared method

Reply:
Done